src/Javis.jl

```diff
@@ -67,32 +71,158 @@ end
 abstract type Transition end
 abstract type InternalTransition end
 
-abstract type ActionType end
+abstract type AbstractAction end
+
+"""
+    SubAction <: AbstractAction
+
+Defines a subaction with relative frames and a function.
+
+# Fields
+- `frames::UnitRange{Int}`: the frames relative to the parent [`Action`](@ref)
+- `func::Function`: the function that gets called in each of those frames.
+    Takes the following arguments: video, action, subaction, rel_frame
+"""
+mutable struct SubAction <: AbstractAction
+    frames                  :: UnitRange{Int}
+    func                    :: Function
+    transitions             :: Vector{Transition}
+    internal_transitions    :: Vector{InternalTransition}
+end
+
+"""
+    SubAction(frames::UnitRange, func::Function)
+
+A `SubAction` can be defined with frames and a function
+inside the `subactions` kwarg of an [`Action`](@ref).
+In the following example the `house_of_nicholas` (which is not part of Javis)
+appears in the first 20 frames which means the opacity is increased from 0 to 1.0.
+Then it stays at full opacity and disappears the same way in the last 20 frames.
+
+# Example
+javis(demo, [
+    BackgroundAction(1:100, ground),
+    Action((args...)->house_of_nicholas(); subactions = [
```

Comment:
I think that instead of using the `house_of_nicholas` function, instead, just use the `circ` func we generally use. The reason for this is the `circ` func is small enough to include here and also a straightforward example. If you had not explained the other day what the `house_of_nicholas` was, I personally would have gotten confused by this example and what you were trying to show.

It's a great idea, just maybe not use house of nicholas here. Perhaps in a tutorial or something?

Reply:
Good point!